source/transactions-convenient-api/transactions-convenient-api.rst

```diff
@@ -119,9 +119,13 @@ operations with the transaction; however, that is not enforced. The callback is
 allowed to execute other operations not associated with the transaction.
 
 Since ``withTransaction`` includes logic to retry transactions and commits,
-drivers MUST enforce a 120-second timeout to limit retry behavior and safeguard
-applications from long-running (or infinite) retry loops. Drivers SHOULD use a
-monotonic clock to determine elapsed time.
+drivers MUST apply timeouts per `Client Side Operations Timeout: Convenient
+Transactions API
+<../client-side-operations-timeout/client-side-operations-timeout.rst#convenient-transactions-api>`__.
+If ``timeoutMS`` is unset for a ``withTransaction`` call, drivers MUST
```

Comment:
Is `timeoutMS` an additional transaction option?  I'm not clear where you specify it for `withTransaction`

Reply:
The [ClientSession Changes](https://github.com/mongodb/specifications/blob/master/source/transactions-convenient-api/transactions-convenient-api.rst#clientsession-changes) section of the convenient transactions spec defines `withTransaction` to take "other arguments as needed", so I assumed `timeoutMS` would go there. I don't think we want to make it a transaction option because that would affect transactions run via the regular API as well.

Reply:
So to take Java as an example, we have:

    <T> T withTransaction(TransactionBody<T> transactionBody);
    <T> T withTransaction(TransactionBody<T> transactionBody, TransactionOptions options);

and the proposal is to add:

    <T> T withTransaction(TransactionBody<T> transactionBody, long timeoutMS);
    <T> T withTransaction(TransactionBody<T> transactionBody, TransactionOptions options, long timeoutMS);

This seems rather odd.  Are there no other options that are feasible (no pun intended)?

Reply:
Spoke offline about this and decided to add a `WithTransactionOptions` type. I added this in my more recent commits both here and in other specs that required it.